core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala

```diff
@@ -633,7 +633,9 @@ object JsonCodec {
   implicit val kubeClusterStatusDecoder: Decoder[KubernetesClusterStatus] = Decoder.decodeString.emap(s =>
     KubernetesClusterStatus.stringToObject.get(s).toRight(s"Invalid cluster status ${s}")
   )
-  implicit val appSamIdDecoder: Decoder[AppSamResourceId] = Decoder.decodeString.map(AppSamResourceId)
+  implicit val appSamIdDecoder: Decoder[AppSamResourceId] =
+    Decoder.forProduct2("resourceId", "accessScope")(AppSamResourceId.apply)
```

Comment:
sbt is yelling at me because of:

```
could not find implicit value for parameter decodeA1: io.circe.Decoder[Option[org.broadinstitute.dsde.workbench.leonardo.AppAccessScope]]
Decoder.forProduct2("resourceId", "accessScope")(AppSamResourceId.apply)
```

Fixing this depends on whether my new implementation of the AppSamResourceId makes sense